include/unifex/v1/debug_async_scope.hpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/just_from.hpp>
+#include <unifex/on.hpp>
+#include <unifex/spawn_detached.hpp>
+#include <unifex/spawn_future.hpp>
+#include <unifex/v1/async_scope.hpp>
+#include <unifex/detail/debug_async_scope.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+inline namespace v1 {
+
+namespace _debug_async_scope {
+
+struct debug_async_scope final {
+  template <typename Sender>
+  auto spawn(Sender&& sender) {
+    return spawn_future(static_cast<Sender&&>(sender), *this);
+  }
+
+  template(typename Sender, typename Scheduler)  //
+      (requires scheduler<Scheduler>)            //
+      auto spawn_on(Scheduler&& scheduler, Sender&& sender) {
+    return spawn(
+        on(static_cast<Scheduler&&>(scheduler), static_cast<Sender&&>(sender)));
+  }
+
+  template(typename Scheduler, typename Fun)             //
+      (requires scheduler<Scheduler> AND callable<Fun>)  //
+      auto spawn_call_on(Scheduler&& scheduler, Fun&& fun) {
+    return spawn_on(
+        static_cast<Scheduler&&>(scheduler),
+        just_from(static_cast<Fun&&>(fun)));
+  }
+
+  template <typename Sender>
+  auto detached_spawn(Sender&& sender) {
+    spawn_detached(static_cast<Sender&&>(sender), *this);
+  }
+
+  template(typename Sender, typename Scheduler)  //
+      (requires scheduler<Scheduler>)            //
+      auto detached_spawn_on(Scheduler&& scheduler, Sender&& sender) {
+    detached_spawn(
+        on(static_cast<Scheduler&&>(scheduler), static_cast<Sender&&>(sender)));
+  }
+
+  template(typename Scheduler, typename Fun)             //
+      (requires scheduler<Scheduler> AND callable<Fun>)  //
+      void detached_spawn_call_on(Scheduler&& scheduler, Fun&& fun) {
+    detached_spawn_on(
+        static_cast<Scheduler&&>(scheduler),
+        just_from(static_cast<Fun&&>(fun)));
```

Comment:
This should probably `static_assert` that `fun` is no-throw callable; if `fun` throws, it'll invoke `terminate()` on completion.